hw/ip/otbn/dv/rig/rig/gens/edge_load_store.py

```diff
@@ -0,0 +1,243 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import random
+from typing import Optional
+
+from shared.operand import ImmOperandType, RegOperandType, OptionOperandType
+from shared.insn_yaml import Insn, InsnsFile
+
+from ..config import Config
+from ..model import Model
+from ..program import ProgInsn, Program
+
+from ..snippet import ProgSnippet
+from ..snippet_gen import GenCont, GenRet, SnippetGen
+
+
+class EdgeLoadStore(SnippetGen):
+    '''A snippet generator that generates random stores that try to
+    hit edge cases.
+
+    '''
+    def __init__(self, cfg: Config, insns_file: InsnsFile) -> None:
+        super().__init__()
+
+        self.insns = []
+        self.weights = []
+
+        self.sw = self._get_named_insn(insns_file, 'sw')
+        self.bnsid = self._get_named_insn(insns_file, 'bn.sid')
+
+        for insn in [self.sw, self.bnsid]:
+            weight = cfg.insn_weights.get(insn.mnemonic)
+            if weight > 0:
+                self.weights.append(weight)
+                self.insns.append(insn)
+
+        # Check that at least one instruction has a positive weight
+        assert len(self.insns) == len(self.weights)
+        if not self.weights:
+            self.disabled = True
+
+    def gen(self,
+            cont: GenCont,
+            model: Model,
+            program: Program) -> Optional[GenRet]:
+
+        self.dmem_size = program.dmem_size
```

Comment:
As above, this could probably just be a local variable and doesn't need to be a field on the generator object.